### PR TITLE
BuildTools: Makefile needs Darwin-specific paths

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

Turns out OSX is a little different from Linux.

`sed` for example in default OSX expects different syntax for `-i`

We could require installing `gnu-sed` but this seems easier.

I added some bash logic to check the current OS name and perform the dependency sync from there.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,15 +33,24 @@ sync-minio-version:
...
-	@echo "Replacing Variables"
-
 	@cp source/default-conf.py source/conf.py
 
-	@sed -i "" "s|MINIOLATEST|${MINIO}|g" source/conf.py
-	@sed -i "" "s|DEBURL|${DEB}|g" source/conf.py
-	@sed -i "" "s|RPMURL|${RPM}|g" source/conf.py
-
-	@git add source/conf.py && git commit -m "Updating MinIO server to ${MINIO}"
+	@if [ "$(shell uname)" == "Darwin" ]; then \
```
Comment (harshavardhana) [inline Makefile:38]:
```suggestion
        @kname=$(uname -s)
        @case "${kname}" in \
        Darwin )
        	sed -i "" "s|MINIOLATEST|${MINIO}|g" source/conf.py; \
		sed -i "" "s|DEBURL|${DEB}|g" source/conf.py; \
		sed -i "" "s|RPMURL|${RPM}|g" source/conf.py; \
		;; \
	*) \
        	sed -i "s|MINIOLATEST|${MINIO}|g" source/conf.py; \
		sed -i "s|DEBURL|${DEB}|g" source/conf.py; \
		sed -i "s|RPMURL|${RPM}|g" source/conf.py; \
                ;; \
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (ravindk89) [reply Makefile]: Whoops. Munged the suggestion, will fix this. Thanks for the suggestion.

File: Makefile [hunk 2]
```diff
@@ -35,7 +35,18 @@ sync-minio-version:
...
+        	sed -i "" "s|MINIOLATEST|${MINIO}|g" source/conf.py; \
+		sed -i "" "s|DEBURL|${DEB}|g" source/conf.py; \
+		sed -i "" "s|RPMURL|${RPM}|g" source/conf.py; \
+		;; \
+	*) \
+        	sed -i "s|MINIOLATEST|${MINIO}|g" source/conf.py; \
+		sed -i "s|DEBURL|${DEB}|g" source/conf.py; \
+		sed -i "s|RPMURL|${RPM}|g" source/conf.py; \
+                ;; \
 		sed -i "" "s|MINIOLATEST|${MINIO}|g" source/conf.py; \
```
Comment (harshavardhana) [inline Makefile:50]: This section must be gone.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by ravindk89** [comment]
@djwfyi when you get a chance can you pull this into your lcal build and see if you hit any more errors? If not then I can merge this and integrate it into the IA docs.

@harshavardhana thanks for your help here

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**